--- a/newlib/libc/machine/xtensa/strcmp.S	2015-11-12 16:23:00.623220071 +0000
+++ b/newlib/libc/machine/xtensa/strcmp.S	2015-11-12 18:00:14.995375550 +0000
@@ -26,6 +26,8 @@
 #define MASK4 0x40404040
 
 
+	.align	4
+	.literal_position
 #if XCHAL_HAVE_L32R
 	.literal .Lmask0, MASK0
 	.literal .Lmask1, MASK1
@@ -33,10 +35,6 @@
 	.literal .Lmask3, MASK3
 	.literal .Lmask4, MASK4
 #endif /* XCHAL_HAVE_L32R */
-
-	.text
-	.align	4
-	.literal_position
 	.global	strcmp
 	.type	strcmp, @function
 strcmp:
