This patch fixes vendor SDK header "c_types.h" for C99 compatibility.

--- ESP8266_NONOS_SDK_V2.0.0_16_07_19/include/c_types.h.orig	2016-07-18 07:16:07.000000000 -0700
+++ ESP8266_NONOS_SDK_V2.0.0_16_07_19/include/c_types.h	2016-07-26 21:45:18.904657700 -0700
@@ -6,17 +6,19 @@
 #ifndef _C_TYPES_H_
 #define _C_TYPES_H_
 
-typedef unsigned char       uint8_t;
+#include <stdint.h>
+#include <stdbool.h>
+//typedef unsigned char       uint8_t;
 typedef signed char         sint8_t;
-typedef signed char         int8_t;
-typedef unsigned short      uint16_t;
+//typedef signed char         int8_t;
+//typedef unsigned short      uint16_t;
 typedef signed short        sint16_t;
-typedef signed short        int16_t;
-typedef unsigned int        uint32_t;
+//typedef signed short        int16_t;
+//typedef unsigned int        uint32_t;
 typedef signed long         sint32_t;
-typedef signed int          int32_t;
+//typedef signed int          int32_t;
 typedef signed long long    sint64_t;
-typedef unsigned long long  uint64_t;
+//typedef unsigned long long  uint64_t;
 typedef unsigned long long  u_int64_t;
 typedef float               real32_t;
 typedef double              real64_t;
@@ -82,10 +84,10 @@
 #define STORE_ATTR __attribute__((aligned(4)))
 
 #ifndef __cplusplus
-typedef unsigned char   bool;
+//typedef unsigned char   bool;
 #define BOOL            bool
-#define true            (1)
-#define false           (0)
+//#define true            (1)
+//#define false           (0)
 #define TRUE            true
 #define FALSE           false
 
